tests/components/met/test_config_flow.py

```diff
@@ -130,3 +132,28 @@ async def test_onboarding_step_abort_no_home(
 
     assert result["type"] == "abort"
     assert result["reason"] == "no_home"
+
+
+async def test_show_options_form(hass: HomeAssistant) -> None:
+    """Test show options form."""
+    entry = await init_integration(hass, track_home=True)
+
+    result = await hass.config_entries.options.async_init(entry.entry_id)
+
+    assert result["type"] == "form"
+    assert result["step_id"] == "init"
+
+
+async def test_options_update_config_entry(hass: HomeAssistant) -> None:
+    """Test options flow updating config entry."""
+    entry = await init_integration(hass, track_home=True)
+
+    update_data = {"name": "test", "latitude": 12, "longitude": 23, "elevation": 456}
+
+    result = await hass.config_entries.options.async_init(
+        entry.entry_id, data=update_data
+    )
+
+    assert result["type"] == "create_entry"
+    assert result["title"] == "Mock Title"
+    assert result["data"] == update_data
```

Comment:
These tests should be combined to one, where the options flow is initialized with empty data and user data is then passed in. In that way it better mimics the real behavior when the user configures the flow.

Also, the test should assert the API is called with the new location, maybe something like this:

```py
    with patch("homeassistant.components.met.metno.MetWeatherData") as weatherdatamock, patch(
        "homeassistant.components.met.metno.MetWeatherData.fetching_data",
        return_value=True,
    ):
        result = await hass.config_entries.options.async_init(
            entry.entry_id, data=update_data
        )
        await hass.async_block_till_done()
    weatherdatamock.assert_called_once_with(location, ANY, ANY)
```